File: src/main/java/mnm/mods/tabbychat/api/ChannelStatus.java
package mnm.mods.tabbychat.api;

/**
 * An enum for chanel statuses. A null value means the channel is currently
 * inactive. Only one status can be in effect at a time.
 * <p>
 * Setting a status with a lower priority than the current one will not take
 * effect. Priority is determined by the ordinal; a lower ordinal has a higher
 * priority. e.g. {@link #ACTIVE} has the highest priority while {@link #JOINED}
 * has the lowest.
 * <p>
 * {@code null} is special and has both highest and lowest priorities. The
 * status will always be set when the old value or new value is null.
 */
public enum ChannelStatus {

    /**
     * For the active channel.
     */
    ACTIVE,
    /**
     * For when your name is mentioned.
     */
    PINGED,
    /**
     * For unread messages.
     */
    UNREAD,
    /**
     * For when players join or leave the channel.
     */
    JOINED
}

File: src/main/kotlin/mnm/mods/tabbychat/api/ChannelStatus.kt
package mnm.mods.tabbychat.api

/**
 * An enum for chanel statuses. A null value means the channel is currently
 * inactive. Only one status can be in effect at a time.
 *
 *
 * Setting a status with a lower priority than the current one will not take
 * effect. Priority is determined by the ordinal; a lower ordinal has a higher
 * priority. e.g. [.ACTIVE] has the highest priority while [.JOINED]
 * has the lowest.
 *
 *
 * `null` is special and has both highest and lowest priorities. The
 * status will always be set when the old value or new value is null.
 */
enum class ChannelStatus {

    /**
     * For the active channel.
     */
    ACTIVE,
    /**
     * For when your name is mentioned.
     */
    PINGED,
    /**
     * For unread messages.
     */
    UNREAD,
    /**
     * For when players join or leave the channel.
     */
    JOINED
}
